Pinta.Effects/Adjustments/BrightnessContrastEffect.cs

```diff
@@ -51,22 +37,23 @@ public override void LaunchConfiguration ()
 
 	public override void Render (ImageSurface src, ImageSurface dest, Core.RectangleI[] rois)
 	{
-		if (!table_calculated)
-			Calculate ();
+		var fraction = GetFraction (Data);
+
+		var rgbTable = CreateRgbTable (fraction.Multiply, fraction.Divide);
 
 		var src_data = src.GetReadOnlyPixelData ();
 		var dst_data = dest.GetPixelData ();
 		int width = src.Width;
 
-		foreach (Core.RectangleI rect in rois) {
+		foreach (var rect in rois) {
 			for (int y = rect.Top; y <= rect.Bottom; y++) {
 				var src_row = src_data.Slice (y * width + rect.Left, rect.Width);
 				var dst_row = dst_data.Slice (y * width + rect.Left, rect.Width);
 
-				if (divide == 0) {
+				if (fraction.Divide == 0) {
 					for (int i = 0; i < src_row.Length; ++i) {
 						ref readonly ColorBgra col = ref src_row[i];
-						uint c = rgb_table![col.GetIntensityByte ()]; // NRT - Set in Calculate
+						uint c = rgbTable![col.GetIntensityByte ()]; // NRT - Set in Calculate
```

Comment:
I think the `!` and the trailing comment can be removed since `rgbTable` is no longer nullable?
Same for other occurrences below

Reply:
You are right about that. I just removed them (2 in total), along with the NRT comment